aziotctl/src/internal/check/checks/well_formed_configs.rs

```diff
@@ -171,14 +189,71 @@ impl Checker for WellFormedIdentitydConfig {
 enum DaemonCfg<T> {
     Cfg(T),
     PermissionDenied(Error),
+    IncorrectPermissions(Error),
+    IncorrectGroupId(Error),
+    IncorrectUserID(Error),
 }
 
 async fn load_daemon_cfg<T: serde::de::DeserializeOwned>(
     daemon: &str,
+    uid: &str,
     config_path: &Path,
     config_directory_path: Option<&Path>,
     shared: &CheckerShared,
 ) -> Result<DaemonCfg<T>> {
+    use std::os::linux::fs::MetadataExt;
+    if let Ok(metadata) = fs::metadata(config_path) {
+        if !metadata.permissions().readonly() {
+            return Ok(DaemonCfg::IncorrectPermissions(anyhow::anyhow!(
+                "The file {} must be read only. Please run 'sudo chmod 0444 {}'",
+                config_path.to_string_lossy(),
+                config_path.to_string_lossy()
+            )));
+        }
+
+        let user = match User::from_uid(Uid::from_raw(metadata.st_uid())) {
+            Ok(Some(user)) => user.name,
+            Err(_) | Ok(None) => {
+                return Err(anyhow::anyhow!(format!(
+                    "Could not find user assigned to {}",
+                    config_path.to_string_lossy()
+                )))
+            }
+        };
+
+        let group = match Group::from_gid(Gid::from_raw(metadata.st_gid())) {
+            Ok(Some(group)) => group.name,
+            Err(_) | Ok(None) => {
+                return Err(anyhow::anyhow!(format!(
+                    "Could not find user group assigned to {}.",
+                    config_path.to_string_lossy()
+                )))
+            }
+        };
+
+        if !user.eq(uid) {
```

Comment:
Compare by UID, not by username.

Reply:
That's straightforward but what's wrong with that?
instead I will have to conver aziotcs (for example) to a uid/gid.